ansible/roles/java-base/tasks/main.yml

```diff
@@ -4,8 +4,7 @@
 # installs java
 #
 
-- name: install java
-  package: name="{{ java_package_name }}" state=present
 # if this fails you want to check in vars/main.yml and add package name
 # as appropriate -- try to use generic os family if available.
-  failed_when: "'{{ java_package_name }}' == 'missing'"
+- name: install java
+  package: name="{{ java_package_name }}" state=present
```

Comment:
I recall the reason I had this in here was that earlier versions of ansible had a bug I was circumventing. Either way, the intended behavior is to stop, so it is an improvement.